File: projects/engineering.purdue.edu/kak/programmingwithobjects/pwocode/chap6/ExplicitCast1.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 6  The Primitive Types and Their Input/Output
//
// Section:     Section 6.7.4  Explicit Type Conversion in Java  
//







//ExplicitCast1.java

import java.io.*;

class Test {
    public static void main( String[] args )
    {
        try {
            PrintWriter out = new PrintWriter( 
                                new FileOutputStream( "out_file" ) );      
            char ch_value;
            for (int i=0; i< 10000; i++) {
                ch_value = (char) i;                              //(A)      
                out.println( "for i= " + i + " char is " + ch_value );
            }
            out.close();
        } catch( IOException e) { }
    }
}

File: projects/engineering.purdue.edu/programmingwithobjects/pwocode/chap6/ExplicitCast1.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 6    The Primitive Types and Their Input/Output
//
// Section:     Section 6.7.3  Explicit Type Conversion in C++
//




//ExplicitCast1.cc

#include <iostream>
using namespace std;

int main()
{
    char ch, ch_prev;

    while (1) {
        cout << "Enter a character: " ;
        cin >> ch;
        if ( ch == ch_prev ) break;                               //(A)      
        ch_prev = ch;
        cout << "Integer value of the character is: " 
             << static_cast<int>(ch) << endl;                     //(B)
    }
    return 0;
}
